### PR TITLE
docs(style-guide): make barrels a consider

### PR DESCRIPTION
**Pull request by Foxandxss**
I changed barrels (as accorded) to consider and while at it, also put the file name as consider as well (for those who don't want to add a new entry to systemjs and prefer to name the barrel like the folder).

Added a new why too.

### CODE REVIEW

File: public/docs/ts/latest/guide/style-guide.jade [hunk 1]
```diff
@@ -1184,6 +1184,10 @@ a(href="#toc") Back to top
 .s-why
   :marked
     **Why?** A barrel reduces the number of imports a file may need.
+    
+.s-why
+  :marked
+    **why?** Without this, we would sometimes import a folder and sometimes import a file.
```
Comment (johnpapa) [inline public/docs/ts/latest/guide/style-guide.jade]: I prefer

> A barrel provides a consistent pattern to import everything exported in the barrel from a folder. 

> This is consistent with a pattern from Node, which imports the `index.js|ts` file from a folder

**Comment by johnpapa** [comment]
revised the *WHY*s

**Comment by Foxandxss** [comment]
Please rereview

**Comment by johnpapa** [comment]
I dont see the changes I suggested

**Comment by johnpapa** [comment]
LGTM

approved

**Pull request closed without merging**